api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java

```diff
@@ -361,47 +361,166 @@ public CohortDefinition getPatientsLTFUMoreThan90DaysComposition() {
   }
 
   /**
-   * Get Patients Transferred out Using the following criteria:
-   * a:patientsWhoLeftARTProgramBeforeOrOnEndDate b:patientsWithMissedVisitOnMasterCard
+   * Get patients who where Transferred Out (and had no registred drug pickup) after the transferred
+   * out date within reporting period
    *
-   * @return
+   * @return CohortDefinition
    */
   public CohortDefinition getTransferredOutPatientsComposition() {
-    CompositionCohortDefinition cd = new CompositionCohortDefinition();
-    cd.setName("Get all patients who are Transferred Out");
-    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
-    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
-    cd.addParameter(new Parameter("location", "Location", Date.class));
 
-    cd.addSearch(
-        "LeftARTProgramBeforeOrOnEndDate",
-        EptsReportUtils.map(
-            getPatientsTransferedOutInProgramBeforeOrOnEndDate(),
-            "onOrBefore=${endDate},location=${location}"));
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
 
-    cd.addSearch(
-        "permanentStateTransferredOut",
-        EptsReportUtils.map(
-            txCurrCohortQueries
-                .getTransferredOutPatientsInFichaResumeAndClinicaOfMasterCardByReportEndDate(),
-            "onOrBefore=${endDate},location=${location}"));
+    sqlCohortDefinition.setName(
```

Comment:
@rubailly So if this is the same as Prev and TB, why not to use the same query?

Reply:
I mean all them from the same place?

Reply:
I think I heard at some point @ynurmahomed mentioning this cohort query classes should be independent as much as possible. I then found the method was duplicated in two classes and thought it was intentional.
I didn't give it much thought but to me having a parent class where they can all inherit these queries would be a better approach. Maybe we can have a discussion around this again with Yassin and come up with a conclusion.

Reply:
@ningosi what do you think about this?

Reply:
Let's have this merged and then we will agree on this!

Reply:
@rubailly @psmatsinhe I think the only re-usable queries are in the [common queries doc ](https://docs.google.com/document/d/1EtpeIn-6seD5skZJteCdANhxkKXQye9RckGV2eoYj6c/edit?pli=1). 

If it does not point to that doc maybe better to duplicate it.

In the past I've tried re-using first then if the specs diverge duplicate it, but ended up having to duplicate the query most of the time.